### PR TITLE
Add import ova for vSphere

### PR DESCRIPTION
**Pull request by jcpowermac**
This PR requires #2922 

Used @jstuever branch as the basis of this change.

`import.go` implements the ability for the installer to
import an RHCOS OVA into vSphere.
`findImportOvaParams` will only provide a HostSystem
that contains the Network and Datastore provided for
the import process.

Used `govc/importx/archive.go` which is required
to retrieve the Ovf Envelope

* [ ]  Rebase once #2922 merges
* [x]  "Template" naming convention
* [ ]  Logging
* [x]  Progress?
* [x]  Additional testing

### CODE REVIEW

File: pkg/asset/cluster/vsphere/import.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+			}
+		}
+		for _, nMoRef := range hostSystemManagedObject.Network {
+			if importOvaParams.Network.Reference().Value == nMoRef.Value {
+				foundNetwork = true
+				break
+			}
+		}
+
+		if foundDatastore && foundNetwork {
```
Comment (jcpowermac) [inline pkg/asset/cluster/vsphere/import.go]: TODO: return error if either are not found have looping through `HostSystem(s)`

**Comment by abhinavdahiya** [comment]
> https://github.com/openshift/installer/commit/bcd5c62fcfd5f0d81757094f07afaaad7b6ac34e

Most of this should be made private or internal package imo.

File: pkg/asset/cluster/vsphere/archive.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+// only encapsulates some common archive related functionality.
+type ArchiveFlag struct {
+	Archive
+}
+
+func newArchiveFlag(ctx context.Context) (*ArchiveFlag, context.Context) {
+	return &ArchiveFlag{}, ctx
+}
+
+// Register ArchiveFlag
```
Comment (patrickdillon) [inline pkg/asset/cluster/vsphere/archive.go]: It seems like the only thing being changed from the [govmomi repo](https://raw.githubusercontent.com/vmware/govmomi/d020dc1638f5e609887634e720040394db7a2ac2/govc/importx/archive.go) is adding some comments. These are exported types so I would think we should be able to use these just from the vendored code and not copy here...

Comment (jstuever) [reply pkg/asset/cluster/vsphere/archive.go]: It's a technical reason... the things we need from the vendored code is not exportable. Therefore, we copied file(s) here with a modified `package vsphere` thereby making them accessible.

**Pull request closed without merging**